src/explorer/Components/MinMaxRefiner.cs

```diff
@@ -15,16 +16,38 @@ public class MinMaxRefiner : ExplorerComponent<MinMaxRefiner.Result>, PublisherC
         private const int MaxIterations = 10;
         private readonly DConnection conn;
         private readonly ExplorerContext ctx;
+        private readonly ResultProvider<MinMaxFromHistogramComponent.Result> histogramMinMaxProvider;
 
-        public MinMaxRefiner(DConnection conn, ExplorerContext ctx)
+        public MinMaxRefiner(
+            DConnection conn,
+            ExplorerContext ctx,
+            ResultProvider<MinMaxFromHistogramComponent.Result> histogramMinMaxProvider)
         {
             this.ctx = ctx;
+            this.histogramMinMaxProvider = histogramMinMaxProvider;
             this.conn = conn;
         }
 
+        public async IAsyncEnumerable<ExploreMetric> YieldMetrics()
+        {
+            var bounds = await ResultAsync;
+
+            if (!(bounds is null))
+            {
+                yield return new UntypedMetric("min", bounds.Min, priority: 10);
+                yield return new UntypedMetric("max", bounds.Max, priority: 10);
+            }
+        }
+
         protected override async Task<Result> Explore()
         {
-            return new Result(await RefinedMinEstimate(), await RefinedMaxEstimate());
+            var histogramBounds = await histogramMinMaxProvider.ResultAsync;
+
+#pragma warning disable CS8603 // Possible null reference return
+            return histogramBounds is null
+                ? new Result(await RefinedMinEstimate(), await RefinedMaxEstimate())
+                : null;
+#pragma warning restore CS8603 // Possible null reference return
```

Comment:
It is not clear why you disable the warning here.

Reply:
I disabled it because it's warning me that I am 'possibly' returning `null` even though I am returning `null` explicitly and on purpose.

Reply:
But why you return `null` when the return type is not nullable? I think this may cause some problems.

Reply:
I've encoded the result status explicitly in the class

112a9be